### ISSUE TITLE
Bug: Storybook for About Modal does not allow for `open` of modal

### ISSUE THREAD

**Issue by sydrosa** [carbon-design-system/ibm-cloud-cognitive#638]
## Unintended behavior in `AboutModal` storybook

Looks like currently the docgen for `AboutModal` shows a boolean for the `open` prop -- this doesn't allow the modal to be opened when pressing the trigger button and can be quite confusing for demo purposes.

CC @dcwarwick for vis. Anyone can pick up this bug!

**Comment by dcwarwick** [comment]
I looked through AboutModal, and thought about several of the props, including `open`. It seems to me there are several cases to consider:
* when a prop can usefully be directly edited, we should ensure it shows a useful control, which may mean overriding the default control sometimes.
* when a prop cannot be directly edited, _e.g._ when it takes an array or object structure or DOM fragment, we should provide a select of useful options.
* when a prop should not be edited at all, because we've provided another way to use it (_e.g._ the `open` prop for AboutModal, where we provide an open button and handle close behaviors), we should disable the control in the controls panel to avoid conflicts or confusion.
* when a prop cannot be edited at all, _e.g._ handlers, we should disable the control in the controls panel.
I've tried to make sure the props are logical in the AboutModal stories in PR #651 -- @sydrosa do take a look and see if there's anything else you'd suggest for how the stories present. I've tried to add some notes to the code guidelines too to capture this.
